src/core/external-calendar-cache.ts

```diff
@@ -0,0 +1,286 @@
+    const entry = this.cache.get(key);
+    
+    if (!entry) {
+      this.stats.misses++;
+      return null;
+    }
+
+    // Check expiration unless explicitly requesting expired entries
+    if (!options.includeExpired && this.isEntryExpired(entry.data)) {
+      this.stats.misses++;
+      return null;
+    }
+
+    // Update access time for LRU
+    entry.lastAccessed = Date.now();
+    this.stats.hits++;
+    
+    return entry.data;
+  }
+
+  /**
+   * Check if cache entry exists
+   */
+  has(key: string): boolean {
+    const entry = this.cache.get(key);
+    return entry !== undefined && !this.isEntryExpired(entry.data);
+  }
+
+  /**
+   * Delete cache entry
+   */
+  delete(key: string): boolean {
+    const deleted = this.cache.delete(key);
+    if (deleted) {
+      this.updateStats();
+    }
+    return deleted;
+  }
+
+  /**
+   * Clear all cache entries
+   */
+  clear(): void {
+    this.cache.clear();
+    this.stats = { hits: 0, misses: 0, totalBytes: 0 };
+    Logger.debug('External calendar cache cleared');
+  }
+
+  /**
+   * Get cache size (number of entries)
+   */
+  size(): number {
+    return this.cache.size;
+  }
+
+  /**
+   * Check if specific cache entry is expired
+   */
+  isExpired(key: string): boolean {
+    const entry = this.cache.get(key);
+    return entry ? this.isEntryExpired(entry.data) : true;
+  }
+
+  /**
+   * Clean up expired entries
+   */
+  cleanupExpired(): void {
+    let cleaned = 0;
+
+    for (const [key, entry] of this.cache.entries()) {
+      if (this.isEntryExpired(entry.data)) {
+        this.cache.delete(key);
+        cleaned++;
+      }
+    }
+
+    if (cleaned > 0) {
+      this.updateStats();
+      Logger.debug(`Cleaned up ${cleaned} expired cache entries`);
+    }
+  }
+
+  /**
+   * Get cache statistics
+   */
+  getStats(): CacheStats {
+    const totalRequests = this.stats.hits + this.stats.misses;
+    const hitRate = totalRequests > 0 ? this.stats.hits / totalRequests : 0;
+
+    return {
+      totalCached: this.cache.size,
+      hits: this.stats.hits,
+      misses: this.stats.misses,
+      hitRate,
+      sizeBytes: this.stats.totalBytes
+    };
+  }
+
+  /**
+   * Configure cache settings
+   */
+  configure(config: Partial<CacheConfiguration>): void {
+    const oldMaxSize = this.config.maxSize;
+    
+    this.config = { ...this.config, ...config };
+
+    // If max size changed, enforce new limit
+    if (config.maxSize !== undefined && config.maxSize !== oldMaxSize) {
+      this.enforceMaxSize();
+    }
+
+    Logger.debug('External calendar cache configured', this.config);
+  }
+
+  /**
+   * Get current configuration
+   */
+  getConfiguration(): CacheConfiguration {
+    return { ...this.config };
+  }
+
+  /**
+   * Check if cached ETag matches provided ETag
+   */
+  hasValidETag(key: string, etag: string): boolean {
+    const entry = this.cache.get(key);
+    return entry?.data.etag === etag;
```

Comment:
I almost wonder if we could count on browser caching too, just something that etag makes me think of.